iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/node/WALNode.java

```diff
@@ -457,8 +455,8 @@ private void snapshotMemTable(DataRegion dataRegion, File tsFile, MemTableInfo m
           "CheckpointManager$DeleteOutdatedFileTask.snapshotOrFlushOldestMemTable");
       try {
         // make sure snapshot is made before memTable flush operation
-        synchronized (memTable) {
-          if (memTable.getFlushStatus() != FlushStatus.WORKING) {
+        synchronized (this) {
```

Comment:
sync this class, Does the WAL write performance be affected?

Reply:
Your question makes sense. I changed it to memTableInfo